### PR TITLE
pdfstudio: added version 2024.0.0

### PR DESCRIPTION
**Pull request by daudi**
Also modified how it is called in all-packages.nix to pass the by-name check.

## Description of changes

Added new 2024 version. https://kbpdfstudio.qoppa.com/whats-new-in-pdf-studio-2024/

Also changed the way that the packages are called, to pass the by-name check. See the [README](https://github.com/NixOS/nixpkgs/blob/614d4b0abbcc78acc950fa8fbca375cb341834e4/pkgs/by-name/README.md#recommendation-for-new-packages-with-multiple-versions)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by daudi** [comment]
@PhilippWoelfel I have added pdfstudio2024. I needed to change the way it is included in all-packages.nix to pass the by-name check.

**Comment by moleculezz** [comment]
How long does it take before PRs are merged?
I have a 2024 license and it doesn't work on 2023.

**Review by PhilippWoelfel — APPROVED** [review]
Looks good to me.

**Comment by PhilippWoelfel** [comment]
> @PhilippWoelfel I have added pdfstudio2024. I needed to change the way it is included in all-packages.nix to pass the by-name check.

Thanks! Do you want to add yourself as maintainer?

**Comment by PhilippWoelfel** [comment]
> Or is there another way to pull in this change?

Yes, there is, but this should probably be discussed in https://discourse.nixos.org/

**Comment by PhilippWoelfel** [comment]
> Or is there another way to pull in this change?

Since someone else asked me how to install this while this has not been merged, here is how you can do this if you're using flakes:
Add repo of the pull request to your flake inputs:
```
inputs = {
  pdfstudioFlake.url = "github:daudi/nixpkgs/pdfstudio2024b";
};
```

And then in your nixos config, you can add this to your system packages:
```
{ pdfstudioFlake, ... }:
{
environment.systemPackages = [ (import pdfstudio2024 { system = "x86_64-linux"; config.allowUnfree = true; }).pdfstudio2024 ];
<other stuff>
}
```

**Comment by moleculezz** [comment]
> ```
> { pdfstudioFlake, ... }:
> {
>   environment.systemPackages = [ (import pdfstudio2024 { system = "x86_64-linux"; config.allowUnfree = true; }).pdfstudio2024 ];
> }
> ```

Should be `[( import pdfstudioFlake { ....`

**Comment by markuskowa** [comment]
There is probably a newer version of 2024 out, but this has been sitting for too long already. Please ping me if you want to update it.

**Pull request merged**
